webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraUserService.scala

```diff
@@ -138,10 +142,20 @@ case class KnoraUserService(
   def removeUserFromGroup(user: KnoraUser, group: Group): IO[UserServiceError, KnoraUser] = for {
     _ <- ZIO
            .fail(UserServiceError(s"User ${user.id.value} is not member of group ${group.groupIri.value}."))
-           .when(!user.isInGroup.contains(group.groupIri))
+           .unless(user.isInGroup.contains(group.groupIri))
     user <- updateUser(user, UserChangeRequest(groups = Some(user.isInGroup.filterNot(_ == group.groupIri)))).orDie
   } yield user
 
+  def removeUserFromKnoraGroup(user: User, group: KnoraGroup): IO[UserServiceError, KnoraUser] =
+    userRepo.findById(user.userIri).someOrFailException.orDie.flatMap(removeUserFromKnoraGroup(_, group))
+
+  def removeUserFromKnoraGroup(user: KnoraUser, group: KnoraGroup): IO[UserServiceError, KnoraUser] = for {
+    _ <- ZIO
+           .fail(UserServiceError(s"User ${user.id.value} is not member of group ${group.id.value}."))
```

Comment:
**question**: Why do we have to fail here?

If a user is not in the group the intention of the caller is fulfilled already so the effect could simple succeed.

Reply:
[d1d97fc](https://github.com/dasch-swiss/dsp-api/pull/3195/commits/d1d97fc1a9aeaba6118f0c7be088207085d24ff5)